plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -67,9 +72,10 @@
     private final Table icebergTable;
     private final Optional<Long> snapshotId;
     private final Map<Integer, Type.PrimitiveType> idToTypeMapping;
-    private final List<Types.NestedField> nonPartitionPrimitiveColumns;
-    private final Optional<RowType> partitionColumnType;
+    private final List<NestedField> nonPartitionPrimitiveColumns;
+    private final Optional<IcebergPartitionColumn> partitionColumnType;
     private final List<io.trino.spi.type.Type> partitionColumnTypes;
```

Comment:
The field `partitionColumnTypes` can be dropped now. 
It is used only once in a code branch which is dependent to `partitionColumnType`